src/main/java/com/example/demo/ShortenUrlController.java

```diff
@@ -1,35 +1,65 @@
 package com.example.demo;
 
-import com.example.demo.dao.UrlRepository;
 import com.example.demo.model.Url;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
 
 @RestController
+@RequestMapping("api/v1/")
 public class ShortenUrlController {
 
     @Autowired
-    private UrlRepository repository;
+    private final ShortenUrlService shortenUrlService;
 
-    @GetMapping
-    public String helloWorld(){
-        return "Hello World from Spring Boot";
+    @Autowired
+    public ShortenUrlController(final ShortenUrlService shortenUrlService){
+        this.shortenUrlService = shortenUrlService;
     }
 
+    //TODO: ShortenUrl Logic
     @PostMapping("/url")
-    public Url saveUrl(@RequestBody Url url){
+    public ResponseEntity<Void> saveUrl(@RequestBody Url url){
+        shortenUrlService.save(url);
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).build();
+    }
 
-        //TODO: Use ResponseEntity
-        return repository.save(url);
+    //TODO: ShortenUrl Logic
+    @GetMapping("/url/{id}")
+    public ResponseEntity<Optional<Url>> findById(@PathVariable Integer id){
+        Optional<Url> urlList = shortenUrlService.findById(id);
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).body(urlList);
     }
 
-    @GetMapping("/url")
-    public List<Url> getUrls(){
+    //TODO: ShortenUrl Logic
+    @DeleteMapping("/url/{id}")
+    public ResponseEntity<Void> deleteById(@PathVariable Integer id){
+        shortenUrlService.deleteById(id);
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).build();
+    }
 
-        //TODO: Use ResponseEntity
-        return repository.findAll();
+    //TODO: ShortenUrl Logic
+    @DeleteMapping("/url")
+    public ResponseEntity<Void> deleteAll(){
+        shortenUrlService.deleteAll();
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).build();
+    }
+
+    //TODO: ShortenUrl Logic
+    @GetMapping("/url")
+    public ResponseEntity<List<Url>> getUrls(){
+        List<Url> urlList = shortenUrlService.findAll();
+        HttpHeaders headers = new HttpHeaders();
+        return ResponseEntity.status(HttpStatus.OK).headers(headers).body(urlList);
     }
 
```

Comment:
As you see, you have to repeat "/url" and repeat headers
Let's have "/url" in a class i.e. `@RequestMapping("api/v1/url")` and have some private method that create Headers once